test/Service/MailServiceTest.php

```diff
@@ -145,6 +147,33 @@ public function whenPreSendReturnsFalseEmailsSendingIsCancelled(): void
         $trigger->shouldHaveBeenCalledTimes(2);
     }
 
+    /** @test */
+    public function whenPreSendReturnsFalseEmailsSendingIsCancelledAndExceptionIsThrown(): void
+    {
+        $mailService = new MailService(
+            $this->transport->reveal(),
+            $this->renderer->reveal(),
+            $this->emailBuilder->reveal(),
+            $this->attachmentParsers->reveal(),
+            $this->eventDispatcher->reveal(),
+            true,
+        );
```

Comment:
You can move the creation of the service to a private method, which only receives the last argument and returns the instance.

That way, it can be called from the `setUp` method and from here, without duplicating the rest.

Somthing in these lines:

```php
private function createMailService(bool $throwOnCancel): MailService
{
    return new MailService(
        $this->transport->reveal(),
        $this->renderer->reveal(),
        $this->emailBuilder->reveal(),
        $this->attachmentParsers->reveal(),
        $this->eventDispatcher->reveal(),
        $throwOnCancel,
    );
}
```